src/main/java/com/SAS/User/Player.java

```diff
@@ -111,4 +114,28 @@ public Team getTeam() {
     public void setTeam(Team team) {
         this.team = team;
     }
+
+    /**
+     * This function remove the asset from the team
+     */
+    @Override
+    public void removeAssetFromTeam() {
+        team.removePlayer(this);
+        this.team = null;
+    }
+
+    /**
+     * This function edits the player details
+     * @param details
+     */
+    @Override
+    public void editDetails(List<String> details) {
+        //first is dateOfBirth, second is fieldRole
+        setDateOfBirth(LocalDate.parse(details.get(0)));
+        setFieldRole(convertStringToFieldRole(details.get(1)));
```

Comment:
convertStringToFieldRole can return null. maybe you need to check it before you set the field role.

Reply:
ok, will check it